homeassistant/components/iron_os/coordinator.py

```diff
@@ -130,6 +125,30 @@ def has_tip(self) -> bool:
             return self.data.live_temp <= threshold
         return False
 
+    async def _update_device_info(self) -> None:
+        """Update device info.
+
+        device info is cached and won't be refetched on every
+        coordinator refresh, only after the device has disconnected
+        the device info is refetched.
+        """
+        build = self.device_info.build
+        self.device_info = await self.device.get_device_info()
+
+        if build != self.device_info.build:
+            device_registry = dr.async_get(self.hass)
+            if TYPE_CHECKING:
+                assert self.config_entry.unique_id
+            device = device_registry.async_get_device(
+                connections={(CONNECTION_BLUETOOTH, self.config_entry.unique_id)}
+            )
+            if device is not None:
+                device_registry.async_update_device(
+                    device_id=device.id,
+                    sw_version=self.device_info.build,
+                    serial_number=f"{self.device_info.device_sn} (ID:{self.device_info.device_id})",
+                )
```

Comment:
Nitpick: You can rewrite this piece of code to return early twice, which will prevent the nesting. 

```
if build == self.device_info.build:
    return

device_registry = dr.async_get(self.hass)
if TYPE_CHECKING:
      assert self.config_entry.unique_id
device = device_registry.async_get_device(
      connections={(CONNECTION_BLUETOOTH, self.config_entry.unique_id)}
            )
(...)
```